irr/src/CImage.cpp

```diff
@@ -95,6 +95,7 @@ SColor CImage::getPixel(u32 x, u32 y) const
 	case ECF_A8R8G8B8:
 		return ((u32 *)Data)[y * Size.Width + x];
 	case ECF_R8G8B8: {
+		// FIXME this is byte order RGB8, whereas SColor is endianness order ARGB8.
```

Comment:
and why is this a problem?

Reply:
It's an inconsistency waiting to shoot someone in the foot.

ARGB8 is "endianness order". It's BGRA8 memory order on a little endian machine, ARGB8 only on a big endian one.

But RGB8 inconsistently is memory order, regardless of whether you're on little or big endian. This is just waiting to cause a red-blue swap if someone uses this carelessly / in the same way as ARGB8.

Reply:
I get the general problem, but there's nothing to fix with this particular piece of code.
It reads the components from memory in a well-defined way and uses the `SColor` component-wise constructor. How `SColor` internally stores the color is not relevant and there's nothing that can go wrong here.

Reply:
> [...] there's nothing to fix with this particular piece of code

The general problem goes beyond this piece of code, but if it is to be fixed, either this piece of code or the ARGB8 getter right above will need to be fixed, because otherwise they are inconsistent:
ARGB8 is 0xAARRGGBB. It depends on endianness. I would expect RGB8 to be 0xRRGGBB, not [R][G][B].
The alternative consistent convention would be for ARGB8 to be [A][R][G][B]. 

> How `SColor` internally stores the color is not relevant

It is, because the ARGB8 -> `SColor` conversion just uses the `u32` constructor and hence depends on endianness and hence is inconsistent with RGB8 true memory order.

---

The best way to fix this, IMO, is to standardize on ARGB8 stored as 0xAARRGGBB in an u32 array and get rid of the rest, and most `CColorConverter` functions, keeping only the ones necessary for X (in memory order) -> ARGB8 conversions for reading images.